### PR TITLE
Support keras model loading

### PR DESCRIPTION
**Pull request by zhichao-li**
## What changes were proposed in this pull request?
It has been verified by __VGG16, VGG19, resnet50 and inceptionV3__
Usage:
``` python
bigdl_model = Model.load_keras(json_path, hdf5_path)
```

## How was this patch tested?
unit test

### CODE REVIEW

**Comment by zhichao-li** [comment]
cc @jason-dai

**Comment by jason-dai** [comment]
Can I load Keras model from Scala?

**Comment by zhichao-li** [comment]
@jason-dai . Currently it only support Python API as we rely on some Keras function to fulfill the job.

File: pyspark/test/bigdl/keras1/test_layer.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        self.modelTestSingleLayer(input_data, layer, functional_api=False)
+
+    def test_batchnormalization(self):
+        # TODO: test training stage result, as the calc logic is not the same for mode 0
+        image_dim_orders = ["th"]
+        modes = ["valid", "same"]
+        for order in image_dim_orders:
+            keras.backend.set_image_dim_ordering(order)
+            print("Testing with %s order" % keras.backend.image_dim_ordering())
+            for mode in modes:
```
Comment (hkvision) [inline pyspark/test/bigdl/keras1/test_layer.py]: Seems that `mode` is never used here?

File: pyspark/bigdl/keras1/converter.py [hunk 2]
```diff
@@ -0,0 +1,1142 @@
...
+                         uRegularizer=self.to_bigdl_reg(config["U_regularizer"]),
+                         bRegularizer=self.to_bigdl_reg(config["b_regularizer"]),
+                         bigdl_type="float")
+        return self.return_sequences(klayer.return_sequences, rec.add(gru))
+
+    def create_batchnormalization(self, klayer, kclayer):
+        config = kclayer["config"]
+
+        self.__check_is_share_weights(kclayer)
+        if keras.backend.image_dim_ordering() != "th" and klayer.axis != 1:
```
Comment (hkvision) [inline pyspark/bigdl/keras1/converter.py]: raise exception for ordering=="th" and klayer.axis != 1?

**Comment by jason-dai** [comment]
Does it support models stored in local file system, HDFS, S3, etc?

**Comment by zhichao-li** [comment]
@jason-dai  issue created to track this: https://github.com/intel-analytics/BigDL/issues/1912
